src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -150,14 +153,20 @@ public interface SymbolLookup {
     @CallerSensitive
     static SymbolLookup loaderLookup() {
         Class<?> caller = Reflection.getCallerClass();
-        ClassLoader loader = Objects.requireNonNull(caller.getClassLoader());
-        MemorySessionImpl loaderSession = MemorySessionImpl.heapSession(loader);
+        // If there's no caller class, fallback to system loader
+        ClassLoader loader = caller != null ?
+                caller.getClassLoader() :
+                ClassLoader.getSystemClassLoader();
+        MemorySession loaderSession = (loader == null) ?
+                MemorySession.global() : // boot loader never goes away
```

Comment:
The other built-in class loaders (`ClassLoaders::appClassLoader` and `ClassLoaders::platformClassLoader` are both instance of `BuiltinClassLoader`) will never be unloaded.   Should they use the globel memory session?

Reply:
good idea